### PR TITLE
[doc] document that `asyncReplace` and `asyncAppend` accept an optional mapper function

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
The `mapper` function optional argument isn't documented. It's shown in the example for `asyncAppend`, but isn't included in the type signature.

This is a documentation only change to document the optional map function.

### Test plan

Look at preview url.

In the first commit I added a more full type, but it overflows the page! Opted to simplify to just `Function` until https://github.com/lit/lit.dev/issues/1293 is fixed.

### CODE REVIEW

File: packages/lit-dev-content/site/docs/v2/templates/directives.md [hunk 1]
```diff
@@ -1831,7 +1831,7 @@ import {asyncAppend} from 'lit/directives/async-append.js';
 <td class="wide-cell">
 
 ```ts
-asyncAppend(iterable: AsyncIterable)
+asyncAppend(iterable: AsyncIterable, mapper?: Function)
```
Comment (augustjk) [inline packages/lit-dev-content/site/docs/v2/templates/directives.md:1834]: I actually like this to be verbose so it shows the index is also provided.
How about doing it multi-line to make it fit?
```suggestion
asyncAppend(
  iterable: AsyncIterable,
  mapper?: (item: Item, index?: number) => unknown
)
```

Comment (AndrewJakubowicz) [reply packages/lit-dev-content/site/docs/v2/templates/directives.md]: Done! Great suggestion!

File: packages/lit-dev-content/site/docs/v2/templates/directives.md [hunk 2]
```diff
@@ -1831,7 +1831,10 @@ import {asyncAppend} from 'lit/directives/async-append.js';
 <td class="wide-cell">
 
 ```ts
-asyncAppend(iterable: AsyncIterable)
+asyncAppend(
+  iterable: AsyncIterable,
+  mapper?: (item: Item, index?: number) => unknown
+)
```
Comment (augustjk) [inline packages/lit-dev-content/site/docs/v2/templates/directives.md:1837]: Sizing now looks good!
One more nit though, I wonder if it should be either
```ts
asyncAppend(
  iterable: AsyncIterable<Item>,
  mapper?: (item: Item, index?: number) => unknown
)
```
or replace `Item` with `T`, or make it `item: unknown` as well?

**Review by augustjk — APPROVED** [review]
Thank you!!

**Pull request merged**